dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/StackTraceJoinerTest.java

```diff
@@ -117,7 +120,39 @@ public void testPrintStackTraceOverrideReflection() throws Exception {
     }
 
     @Test
-    public void testPrintStackTraceMultipleThreads() {
-        // TODO: Test printStackTrace in multiple threads and ensure that messages are not leaking from one stack trace to another
+    public void testPrintStackTraceMultipleThreads() throws Exception {
+        final int threadsToRun = 3;
+        final int expectedErrOut = threadsToRun + 3;
+        Logger logger = Logger.getLogger("com.ibm.ws.logging.stackjoiner");
+        logger.log(Level.INFO, ">>>>> Create threads");
+
+        for (int i = 0; i < threadsToRun; i++) {
+            new Thread(new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                      messagesLog = server.getDefaultLogFile();
+                      server.setMarkToEndOfLog(messagesLog);
```

Comment:
you probably shouldn't be doing this from each thread... would make more sense to do this mark before you spawn the threads.